wakuv2/waku.go

```diff
@@ -1399,16 +1400,21 @@ func (w *Waku) Start() error {
 			case <-peerCountTimer.C:
 				peerCountTimer.Reset(3 * time.Second)
 				newPeerCount := len(w.node.Host().Network().Peers())
-				if newPeerCount != peerCount && w.onPeerStats != nil {
+				if newPeerCount != peerCount && (w.onPeerStats != nil || w.statusTelemetryClient != nil) {
```

Comment:
Why don't we want to set `peerCount` is `onPeerStats` and `statusTelemetryClient` are `nil`? IMO you don't need to check for the `nil` here since you also do it later - or am I missing anything?

Reply:
We update the peer count if either one is true. We only call `onPeerStats` and/or `statusTelemetryClient` if either one is set.